### PR TITLE
build: single source of truth for dependencies

### LINKED ISSUE #107: Improve install procedure
See https://github.com/ComPWA/expertsystem/issues/103 for specification and https://github.com/ComPWA/expertsystem/pull/102 for how to address this issue. This would also be needed for #53.

### PR DESCRIPTION
**Pull request by redeboer**
Closes #107

See also https://github.com/ComPWA/expertsystem/pull/102

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-doc8>=0.8.0
```
Comment (spflueger) [inline doc/requirements.txt:1]: shouldn't theses requirement files be removed now?

Comment (redeboer) [reply doc/requirements.txt:1]: Emm yes... not sure what I was doing there...

Comment (redeboer) [reply doc/requirements.txt:1]: Done: dc2ee290322fd32b36f59929706e47c66cf19c22

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
+    Topic :: Scientific/Engineering
+    Topic :: Scientific/Engineering :: Physics
+    Typing :: Typed
+
+[options]
+python_requires = >=3.6, <3.8
+install_requires =
+    amplitf==0.0a1
```
Comment (spflueger) [inline setup.cfg:35]: we should require pip >= 19 because of tf2

File: environment.yml [hunk 3]
```diff
@@ -6,4 +6,4 @@ dependencies:
   - python>=3.6,<3.8
   - pip
   - pip:
-      - -r requirements.txt
+      - -e .
```
Comment (redeboer) [inline environment.yml:9]: Maybe better use `-e .[dev]`? In that case, the install procedure for contributors using Conda would just become:
```bash
conda env create
conda activate tw
```
which previously would have been
```bash
conda env create
conda activate tw
pip install -e .[dev]
```

But note that this assumes anyone who uses Conda is also interested in contributing (i.e. the dev requirements). And this is not necessarily the case. Could especially be a problem for #53.

Comment (spflueger) [reply environment.yml:9]: hmm i would keep it optional to install the dev requirements. So you can have the run requirements be part of the dev requirements to do it all in one swoop. But im not sure how this is handled in conda

Comment (redeboer) [reply environment.yml:9]: Ok, then I leave it like this (it's also the described workflow in the install/contribute pages)

**Pull request merged**